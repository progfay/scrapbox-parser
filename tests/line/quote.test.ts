/* global describe it expect */

describe('quote', () => {
  it('Simple quote', () => {
    expect('> Simple quote').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'quote',
            nodes: [
              {
                type: 'plain',
                text: ' Simple quote'
              }
            ]
          }
        ]
      }
    ])
  })

  it('Empty quote', () => {
    expect('>').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'quote',
            nodes: []
          }
        ]
      }
    ])
  })
})
