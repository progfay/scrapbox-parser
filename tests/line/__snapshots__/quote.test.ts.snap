// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quote Empty quote 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "nodes": Array [],
        "raw": ">",
        "type": "quote",
      },
    ],
    "type": "line",
  },
]
`;

exports[`quote Simple quote 1`] = `
Array [
  Object {
    "indent": 0,
    "nodes": Array [
      Object {
        "nodes": Array [
          Object {
            "raw": " Simple quote",
            "text": " Simple quote",
            "type": "plain",
          },
        ],
        "raw": "> Simple quote",
        "type": "quote",
      },
    ],
    "type": "line",
  },
]
`;
